Multisig/UI/Assets/AssetsViewModel.swift

```diff
@@ -8,17 +8,37 @@
 
 import Foundation
 import Combine
+import SwiftCryptoTokenFormatter
+import BigInt
 
-struct TokenBalance: Identifiable {
-    var id: String {
-        address
+struct TokenBalance: Identifiable, Hashable {
+    var id: Int {
+        hashValue
     }
-    var imageURL: URL {
-        URL(string: "https://gnosis-safe-token-logos.s3.amazonaws.com/\(address).png")!
+    var imageURL: URL? {
+        // will be replaced when https://github.com/gnosis/safe-transaction-service/issues/86 is ready
+        guard let address = address else { return nil }
+        return URL(string: "https://gnosis-safe-token-logos.s3.amazonaws.com/\(String(describing: address)).png")!
     }
-    let address: String
+    let address: String?
+    let symbol: String
     let balance: String
```

Comment:
As these values are usually parsed from backend and in term of defensive programing, I don't expect that server return values, so I used to make them all optional to avoid cashing on parsing.

Reply:
All these fields are required by the specification. Optional means that the server may return nil value for this field.
I would not use optional everywhere.